### PR TITLE
Fixes #10036 - CRUD bindings for rpm ostree repositories

### PR DESCRIPTION
**Pull request by parthaa**

### CODE REVIEW

File: test/extensions/ostree_repository_test.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+module Extensions
+  module TestOstreeRepositoryBase
+    def setup
+      @support = RepositorySupport.new('ostree')
+      @extension = TestRuncible.server.extensions.repository
+    end
+  end
+
+  class TestOstreeRepositoryCreate < MiniTest::Unit::TestCase
+    include TestOstreeRepositoryBase
```
Comment (ehelms) [inline test/extensions/ostree_repository_test.rb]: Does this need to be an include and separated if there is only this one set of tests?

Comment (parthaa) [inline test/extensions/ostree_repository_test.rb]: @ehelms Just following the template we are using for other kind of repository tests. Who know we may use it to do sync tests

Comment (ehelms) [inline test/extensions/ostree_repository_test.rb]: Heh, planning for the future ehh? Ok.

**Comment by ehelms** [comment]
LGTM

File: test/extensions/ostree_repository_test.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      distributors = [{'type_id' => 'ostree_web_distributor', 'id' => '123', 'auto_publish' => true,
+                       'config' => {'ostree_publish_directory' => '/path'}}]
+      response = @extension.create_with_distributors(RepositorySupport.repo_id, distributors)
+
+      assert_equal 201, response.code
+      assert_equal RepositorySupport.repo_id, response['id']
+    end
+
+    def test_create_with_distributor_object
+      repo_id = RepositorySupport.repo_id + '_distro'
```
Comment (daviddavis) [inline test/extensions/ostree_repository_test.rb]: Favor string interpolation over concatenation.

File: test/extensions/ostree_repository_test.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      ostree_publish_directory = "/path"
+      relative_path = "/relative_path"
+      mock_distro = Runcible::Models::OstreeDistributor.new(:ostree_publish_directory => ostree_publish_directory,
+                                                            :id => '123', :relative_path => relative_path)
```
Comment (daviddavis) [inline test/extensions/ostree_repository_test.rb]: You should probably line this up better.

File: lib/runcible/models/ostree_distributor.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+require 'active_support/json'
+require 'securerandom'
+
+module Runcible
+  module Models
+    class OstreeDistributor < Distributor
+      #optional
```
Comment (daviddavis) [inline lib/runcible/models/ostree_distributor.rb]: This comment is kind of vague. Not sure what it means.

Comment (parthaa) [inline lib/runcible/models/ostree_distributor.rb]: So if you see yum distributor it lists "required" attributes and "optional" attributes. Simiar case here.. Its optional..

Comment (daviddavis) [inline lib/runcible/models/ostree_distributor.rb]: I saw yum distributor. However, I only see one place (yum distributor) where `#required` is being used and I'm guessing if it confused me, it'll likely confuse others.

Comment (parthaa) [inline lib/runcible/models/ostree_distributor.rb]: I think in the rest of the distributors all the required stuff is already there in /runcible/lib/runcible/models/distributor.rb so they do not have anything required.

**Comment by daviddavis** [comment]
@parthaa talked with @jlsherrill and I think we need to make sure all the tests are passing in live mode before we merge this. can you confirm?

**Comment by daviddavis** [comment]
ACK if all tests are passing.

**Comment by daviddavis** [comment]
> @parthaa I talked with @jlsherrill and I think we need to make sure all the tests are passing in live mode before we merge this. can you confirm?

You never confirmed that all tests were passing in live mode.

:-1:

**Comment by parthaa** [comment]
Sorry missed on that comment :). Will look into fixing those soone enough.

**Comment by daviddavis** [comment]
Cool, no rush.

**Pull request merged**